packages/platform-core/src/errors/amplify_error.ts

```diff
@@ -55,7 +55,9 @@ export abstract class AmplifyError extends Error {
     });
   }
 
-  static fromStderr = (_stderr: string): AmplifyError | undefined => {
+  static fromStderr = <T extends string>(
+    _stderr: string
+  ): AmplifyError<T> | undefined => {
```

Comment:
since we aren't actually checking the value of `name` in the error here, the return type should probably just be `AmplifyError<string>`